src/transformers/modeling_gguf_pytorch_utils.py

```diff
@@ -174,6 +174,13 @@ def load_gguf_checkpoint(gguf_checkpoint_path, return_tensors=False):
                 elif ".attn_k." in name:
                     weights = reverse_permute_weights(weights, num_heads, num_kv_heads)
 
+            if architecture == "qwen2moe":
+                if "_exp" in name:
+                    split_moe_expert_tensor(weights, parsed_parameters, name, tensor_key_mapping)
+                    continue
+                if "ffn_gate_inp_shexp" in name:
+                    weights = np.expand_dims(weights, axis=0)
+
```

Comment:
Can you add comments to explain why we need to do these ? I understand the first case but not the second one.

Reply:
Hi @SunMarc! Yes, sure, it is because of tensor shape, it must be (1, 2048) but we got (2048) from quantized one. And then execution is failed on the moment we call `set_module_tensor_to_device(model, param_name, param_device, **set_module_kwargs)` in modeling_utils.py (L989-L991). I added a comment that we do it for compatibility here